xds/src/main/java/io/grpc/xds/XdsClientWrapperForServerSds.java

```diff
@@ -84,41 +84,76 @@ public ManagementServerNotFoundException(String msg) {
   }
 
   /**
-   * Factory method for creating a {@link XdsClientWrapperForServerSds}.
+   * Creates a {@link XdsClientWrapperForServerSds}.
    *
    * @param port server's port for which listener config is needed.
-   * @param bootstrapper {@link Bootstrapper} instance to load bootstrap config.
-   * @param syncContext {@link SynchronizationContext} needed by {@link XdsClient}.
    */
-  public static XdsClientWrapperForServerSds newInstance(
-      int port, Bootstrapper bootstrapper, SynchronizationContext syncContext)
-      throws IOException, ManagementServerNotFoundException {
-    Bootstrapper.BootstrapInfo bootstrapInfo = bootstrapper.readBootstrap();
-    final List<Bootstrapper.ServerInfo> serverList = bootstrapInfo.getServers();
-    if (serverList.isEmpty()) {
-      throw new ManagementServerNotFoundException("No management server provided by bootstrap");
+  public XdsClientWrapperForServerSds(int port) {
+    this.port = port;
+  }
+
+  private SynchronizationContext createSynchronizationContext() {
+    final InternalLogId logId =
+        InternalLogId.allocate("XdsClientWrapperForServerSds", Integer.toString(port));
+    return new SynchronizationContext(
+        new Thread.UncaughtExceptionHandler() {
+          // needed by syncContext
+          private boolean panicMode;
+
+          @Override
+          public void uncaughtException(Thread t, Throwable e) {
+            logger.log(
+                Level.SEVERE,
+                "[" + logId + "] Uncaught exception in the SynchronizationContext. Panic!",
+                e);
+            panic(e);
+          }
+
+          void panic(final Throwable t) {
+            if (panicMode) {
+              // Preserve the first panic information
+              return;
+            }
+            panicMode = true;
+            shutdown();
+          }
+        });
+  }
+
+  /** Creates an XdsClient and starts a watch. */
+  public void start() {
```

Comment:
make sure the start can be called at most once. although it is programmer error, it will be very hard to debug resource leaks and random behavior due to internal reference changes.

Reply:
added the check in `start(XdsClient xdsClient)` which this one calls

Reply:
this is not correct because internal state (timeService) is already mutated.

Reply:
you are right. I will move it to the real "start" then

Reply:
Done. I had to create another `startForTestXdsClient()` method just for the tests.